### PR TITLE
Add porting guide for HAL I2C API

### PR DESCRIPTION
**Pull request by ghost**
This isn't ready to merge until the links are fixed when their target links are merged.

### CODE REVIEW

**Comment by scartmell-arm** [comment]
https://github.com/ARMmbed/mbed-os/pull/8682

**Comment by ashok-rao** [comment]
@AnotherButler : Is this coming for 5.11.0 ?

**Comment by AnotherButler** [comment]
@ashok-rao This is being delivered on a feature branch, not on master.

File: docs/porting/target/i2c.md [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+
+I2C is a serial protocol for two-wire interface to connect low-speed devices
+in embedded systems. The I2C API allows control and configuration of this
+interface.
+
+The interface is made up of two lines for all communication:
+  - SCL: Serial Clock
+  - SDA: Serial Data
+
+### Assumptions
```
Comment (c1728p9) [inline docs/porting/target/i2c.md]: Can assumptions be removed since it is unused?

**Review by c1728p9 — APPROVED** [review]
One minor niptick, but the change looks good to me.

File: docs/porting/target/i2c.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+#### Defined Behaviours
+
+- `i2c_init`:
+  - Initialises the peripheral pins specified in the input parameters.
+  - Initialises the peripheral in master mode if `is_slave` is false.
+  - Initialises the peripheral in slave mode if `is_slave` is true and `supports_slave_mode` is true.
```
Comment (mrcoulter45) [inline docs/porting/target/i2c.md]: Can we add
"- Initializes all object fields."?
The fact that this isn't done I noticed can cause significant problems when calling i2c_init directly without using the higher level I2C driver function (mbed-os issue 9890)

**Comment by AnotherButler** [comment]
New code dependency: https://github.com/ARMmbed/mbed-os/pull/10374

**Comment by maciejbocianski** [comment]
I'm checking it out now

**Comment by maciejbocianski** [comment]
Updated version #1103

**Pull request closed without merging**